src/storage/src/client.rs

```diff
@@ -358,11 +367,16 @@ impl InsertObject {
         );
 
         let builder = self.inner.apply_auth_headers(builder).await?;
-        let content = match self.request.data {
-            Some(Data::ChecksummedData(data)) => data.content,
```

Comment:
comment: ah ok, we used to (internally) hold the payload in the `data` field, but now we hold it separately in something that can be streamed from. Got it.